pkg/io/cryptoReader.go

```diff
@@ -0,0 +1,25 @@
+/*
+	Package io contain custom implementation of io Reader.
+	Using for read data many times without changes.
+*/
+package io
+
+// A RootReader implements the io.Reader
+type rootReader struct {
+	data []byte // data to read
+}
+
+// NewRootReader returns a new Reader reading from p.
+func NewRootReader(p []byte) *rootReader {
```

Comment:
It is not a `RootReader`, better to use a name that tells something about what this reader is doing. For example, `LoopReader` or `RingReader`.

Think also about better description with what it do because `Reader reading from p` is kind of what all readers do.